project-base/app/src/Model/Customer/User/RegistrationDataFactory.php

```diff
@@ -61,4 +64,25 @@ public function createWithArgument(Argument $argument): RegistrationData
 
         return $registrationData;
     }
+
+    /**
+     * @param \Hybridauth\User\Profile $profile
+     * @return \App\Model\Customer\User\RegistrationData
+     */
+    public function createFromSocialNetworkProfile(Profile $profile): RegistrationData
+    {
+        $registrationData = $this->createForDomainId($this->domain->getId());
+        $countries = $this->countryFacade->getAllEnabledOnCurrentDomain();
+
+        $registrationData->firstName = $profile->firstName ?? '';
+        $registrationData->lastName = $profile->lastName ?? '';
+        $registrationData->email = $profile->email;
+        $registrationData->street = '';
+        $registrationData->city = '';
+        $registrationData->postcode = '';
+        $registrationData->country = $countries[0];
```

Comment:
I know this is an edge case but what if there is no enabled country on a domain?

Reply:
as we discussed, It's very little edge case and it will be solved in next follow-up task